### PR TITLE
Update models docs documenting member visibility allowed for the different types of injection

### PR DESCRIPTION
**Pull request by paul-bjorkstrand**
Also, added a subheadings to separate basic usage into two subsections.

### CODE REVIEW

**Comment by paul-bjorkstrand** [comment]
@kwin - here it is.

I'm flexible on the subheadings. I just felt that the manifest part blended in too much with the constructor/class part. Happy to remove them if you feel they are unnecessary.

File: src/main/jbake/content/documentation/bundles/models.md [hunk 1]
```diff
@@ -55,6 +57,35 @@ Constructor injection is also supported (as of Sling Models 1.1.0):
 
 Because the name of a constructor argument parameter cannot be detected via the Java Reflection API a `@Named` annotation is mandatory for injectors that require a name for resolving the injection.
 
+Constructors may use any visibility modifier (as of Sling Models 1.5.0):
```
Comment (kwin) [inline src/main/jbake/content/documentation/bundles/models.md:60]:
```suggestion
Constructors may use any visibility modifier (as of [Sling Models 1.5.0](https://issues.apache.org/jira/browse/SLING-8069)):
```

Comment (kwin) [inline src/main/jbake/content/documentation/bundles/models.md:60]: Shouldn't we mention fields and methods here as well for completion? IMHO those never had any restrictions in terms of visibility....

**Review by kwin — CHANGES_REQUESTED** [review]

File: src/main/jbake/content/documentation/bundles/models.md (hunk 1 above)
Comment (paul-bjorkstrand) [reply src/main/jbake/content/documentation/bundles/models.md:60]: For fields, yes. Call me crazy, but while I was digging in, to try to replace reflection with method handles, I did not see method _injection_ anywhere. Everywhere I see method-injection-like code, it only worked with an interface, using the proxy. I am not sure if private methods in interfaces are proxyable, thus injectable.

I would be happy to be proven wrong, though.

Comment (kwin) [reply src/main/jbake/content/documentation/bundles/models.md]: You are right with regards to methods. Let's just mention fields for now as well.

Comment (paul-bjorkstrand) [reply src/main/jbake/content/documentation/bundles/models.md]: Will make the change and push a bit later today. Regarding method injection for concrete models - I could see that as an enhancement. I'll add a new ticket for that feature

Also, I think the code could use a little attention in `ModeAdapterlFactory`(I REALLY want to refactor it, but just need the time to sit down and understand everything).

Comment (paul-bjorkstrand) [reply src/main/jbake/content/documentation/bundles/models.md]: Updated with a lot more detail re: field and method injection

**Review by kwin — APPROVED** [review]

**Comment by kwin** [comment]
@paul-bjorkstrand Thanks a lot.

**Pull request merged**